output/output.go

```diff
@@ -0,0 +1,134 @@
+// Package output provides an interface for outputting test results in different formats.
+package output
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/kyokomi/emoji"
+	"github.com/rs/zerolog/log"
+)
+
+// Type is a string representing the types of outputs this application has. Each different
+// output will print in a different way, suitable for the defined type.
+type Type string
+
+const (
+	Normal    Type = "normal"
+	Quiet     Type = "quiet"
+	GitHub    Type = "github"
+	GitLab    Type = "gitlab"
+	CircleCI  Type = "circleci"
+	CodeBuild Type = "codebuild"
+	Jenkins   Type = "jenkins"
+	JSON      Type = "json"
+)
+
+type Output struct {
+	oType Type
+	w     io.Writer
+}
+
+func (o *Output) Type() Type {
+	return o.oType
+}
+
+func (o *Output) Println(format string, a ...interface{}) error {
+	err := o.Printf(format+"\n", a...)
+	return err
+}
+
+func (o *Output) Printf(format string, a ...interface{}) error {
```

Comment:
You pass in emoji strings to all output types. Do you need a filter to remove the emoji strings? Or do you want them to appear in the output as `:emojistring:some log text`?

Reply:
Yes, I was thinking on filter emoji. Will filter using regexp.

Reply:
Another option is to have a simplified version of [catalogs](https://pkg.go.dev/golang.org/x/text@v0.4.0/message/catalog) that can be used to map text quickly.

Reply:
Interesting! But maybe a bit overkill for the moment.

Reply:
Simplified this idea.